noxfile.py

```diff
@@ -0,0 +1,73 @@
+"""Nox config."""
+
+from __future__ import annotations
+
+import nox
+
+# Options to modify nox behaviour
+nox.options.default_venv_backend = "uv|virtualenv"
+nox.options.reuse_existing_virtualenvs = True
+nox.options.sessions = ["lint", "tests"]
+
+ALL_PYTHON = ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+
+@nox.session
+def lint(session: nox.Session) -> None:
+    """Run the linter."""
+    session.install("pre-commit")
+    session.run("pre-commit", "run", "--all-files", *session.posargs)
+
+
+@nox.session(python=ALL_PYTHON)
+def tests(session: nox.Session) -> None:
+    """Run the unit tests."""
+    session.install("-c", ".github/test-constraints.txt", "-e", ".[test]")
+    session.run(
+        "pytest",
+        *session.posargs,
+    )
+
+
+@nox.session(python=ALL_PYTHON)
+def coverage(session: nox.Session) -> None:
+    """Run tests and compute coverage."""
+    session.posargs.append("--cov=glass")
+    tests(session)
```

Comment:
Is `tests` nox magic?

Reply:
It is the nox session defined above. The `coverage` session just passes an extra argument to the `tests` session.

Reply:
Ah I see, thanks